CAD-992:  updates for the latest deployment

deepfire opened the pull request.

deepfire committed:
CAD-918 tx-gen: fix the init-time deadlock & improve diagnostics

deepfire committed:
CAD-974 txgen:  config the init<->main delay via --init-cooldown

CodiePP commented on cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx.hs:
```diff
@@ -179,7 +183,11 @@ genesisBenchmarkRunner loggingLayer
-  let (benchTracer, connectTracer, submitTracer, lowLevelSubmitTracer) = createTracers loggingLayer
+  let ( benchTracer
+       , connectTracer
+       , submitMuxTracer
```
alignment off

deepfire replied:
It's not off -- it's a parse error if I align `,` with `(`.

CodiePP replied:
then it might be better to put the comma at the end of the line and align the names?

CodiePP commented on cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx.hs:
```diff
@@ -1022,16 +1046,17 @@ txGenerator benchTracer
   -- Later sumbitter is reading and submitting these transactions.
-  txsForSubmission :: MSTM.TMVar IO [CC.UTxO.ATxAux ByteString] <- liftIO $ STM.newTMVarIO []
+  txsForSubmission :: [MSTM.TMVar IO [CC.UTxO.ATxAux ByteString]] <-
+    liftIO $ replicateM numOfTargetNodes $ STM.newTMVarIO []
```
so this is not only introducing "cooldown"?
doese this still match the comment?

deepfire replied:
This is a part of the fix for the single-threaded submission bug.

..and yes, I think the comment is still true.

CodiePP commented on cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx/Submission.hs:
```diff
@@ -390,34 +393,45 @@ data TraceBenchTxSubmit txid
     case t of
       TraceBenchTxSubRecv _      -> mkObject ["kind" .= A.String "TraceBenchTxSubRecv"]
       TraceBenchTxSubStart _     -> mkObject ["kind" .= A.String "TraceBenchTxSubStart"]
+      TraceBenchTxSubServAnn _  -> mkObject ["kind" .= A.String "TraceBenchTxSubServAnn"]
```
alignment off

deepfire replied:
Done.

CodiePP commented on benchmarks/cluster3nodes/configuration/parameters:
```diff
@@ -6,7 +6,7 @@ numtx=1000
 targetnodes="0 1 2"
```
I think this file should not be changed by the PR

deepfire replied:
Done.

CodiePP reviewed (changes requested):
some comments; needs thorough testing..

deepfire committed:
CAD-992 txgen:  --single-threaded flag

deepfire committed:
CAD-974 txgen:  add a low-volume N2N submission trace

# Conflicts:
#	cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx/Submission.hs

deepfire committed:
CAD-992 review:  address

deepfire commented:
@CodiePP, addressed all comments.

deepfire committed:
CAD-992 scripts:  more log analysis

deepfire committed:
CAD-992 cluster3nodes:  fixes and improvements

deepfire committed:
CAD-992:  adapt to the new log storage scheme

deepfire committed:
CAD-992 | git:  fix the ignore list

deepfire commented:
Closing in favor or componentised PRs.
